src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodCapiDriver.cs

```diff
@@ -0,0 +1,275 @@
+            {
+                eventHandler = new LibgpiodDriverEventHandler(pinNumber, new CancellationTokenSource());
+                _pinNumberToEventHandler.Add(pinNumber, eventHandler);
+                RequestEvent(pinNumber, eventHandler, eventType);
+                InitializeEventDetectionThread(eventHandler);
+            }
+            if (eventType.HasFlag(PinEventTypes.Rising))
+                eventHandler.ValueRising += callback;
+            if (eventType.HasFlag(PinEventTypes.Falling))
+                eventHandler.ValueFalling += callback;
+        }
+
+        private void InitializeEventDetectionThread(LibgpiodDriverEventHandler eventHandler)
+        {
+            Thread pinListeningThread = new Thread(DetectLineEvent)
+            {
+                IsBackground = true
+            };
+
+            pinListeningThread.Start(eventHandler);
+        }
+
+        private void DetectLineEvent(Object obj)
+        {
+            LibgpiodDriverEventHandler eventHandler = (LibgpiodDriverEventHandler)obj;
+            int waitResult = -2;
+            while (!eventHandler.CancellationTokenSource.IsCancellationRequested)
+            {
+                waitResult = Interop.gpiod_line_event_wait(eventHandler.PinHandle, ref _defaultTimeOut);
+                if (waitResult == -1)
+                {
+                    throw new GpioChipException($"Error while waiting for event, error code: {Marshal.GetLastWin32Error()}");
+                }
+                if (waitResult == 1)
+                {
+                    int readResult = Interop.gpiod_line_event_read(eventHandler.PinHandle, out gpiod_line_event eventOccured);
+                    if (readResult == -1)
+                    {
+                        throw new GpioChipException($"Error while trying to read pin event result, error code: {Marshal.GetLastWin32Error()}");
+                    }
+                    PinEventTypes eventType = (eventOccured.event_type == event_type.GPIOD_LINE_EVENT_RISING_EDGE) ? PinEventTypes.Rising : PinEventTypes.Falling;
+                    var args = new PinValueChangedEventArgs(eventType, eventHandler.PinNumber);
+                    eventHandler?.OnPinValueChanged(args, eventType);
+                }
+            }
+        }
+
+        protected internal override void ClosePin(int pinNumber)
+        {
+            if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pin))
+            {
+                Interop.gpiod_line_release(pin);
+                _pinNumberToSafeLineHandle.Remove(pinNumber);
+            }
+        }
+
+        protected internal override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber) => throw new
+            PlatformNotSupportedException("This driver is generic so it cannot perform conversions between pin numbering schemes.");
+
+        protected internal override PinMode GetPinMode(int pinNumber)
```

Comment:
interesting. you somehow broke github coloring 😄

Reply:
😛 is that something need to consider?